visitor.go

```diff
@@ -39,53 +40,56 @@ func (vis *visitor) mustVisit(w io.Writer, v Value) {
 	}
 	defer vis.leave(v)
 
-	cw := count.NewWriter(w)
+	var ok bool
+	if v.Value, ok = unsafereflect.MakeMutable(v.Value); ok {
```

Comment:
Now that this mutable `reflect.Value` is a hard requirement, we'll need to handle the failure of this differently.

Perhaps `MakeMutable()` should just be changed to panic to indicate a failure, rather than returning a boolean.